presto-main/src/main/java/com/facebook/presto/server/protocol/Query.java

```diff
@@ -98,6 +103,8 @@ class Query
 {
     private static final Logger log = Logger.get(Query.class);
     private static final Base64.Encoder BASE64_ENCODER = Base64.getEncoder();
+    private static QueryId originalBeforeRetryQueryId;
```

Comment:
Can there only be one retry ? What if we call this originalQueryId, or perhaps we maintain this as a list (of query ids), where there should be one to begin with but the 0'th item contains the original attempt and then subsequent ones get appended. or would that be an overkill ? 

Same for the plan hashes -- but maybe its ok for the hash, since we only care about the "previous"

Reply:
@agrawaldevesh I think that in this case we only want to check for one query as retries aside from that are for transient errors